packages/frontend/web/components/elements/RichLinkComponent.tsx

```diff
@@ -0,0 +1,259 @@
+`;
+
+const contributorImageWrapper = css`
+    width: 5rem;
+    height: 5rem;
+    margin-left: auto;
+    margin-right: 0.3rem;
+`;
+
+const neutralBackground = css`
+    background-color: ${palette.neutral[97]};
+    a {
+        color: inherit;
+    }
+    :hover {
+        background-color: ${palette.neutral[93]};
+    }
+`;
+
+const pillarBackground: (pillar: Pillar) => colour = pillar => {
+    return css`
+        background-color: ${richLinkPillarColour(pillar)};
+    `;
+};
+
+const textColour: (pillar: Pillar) => colour = pillar => {
+    return css`
+        color: ${richLinkPillarColour(pillar)};
+    `;
+};
+
+const paidForBranding = css`
+    ${textSans(1)};
+    font-weight: bold;
+    color: ${palette.neutral[46]};
+`;
+
+const readMoreText: (contentType: string) => string = contentType => {
+    switch (contentType) {
+        case 'audio':
+            return 'Listen';
+        case 'gallery':
+            return 'View Gallery';
+        case 'video':
+            return 'Watch';
+        default:
+            return 'Read more';
+    }
+};
+
+const getMainContributor: (tags: TagType[]) => string = tags => {
+    const contributorTags = tags.filter(t => t.type === 'Contributor');
+    return contributorTags.length > 0 ? contributorTags[0].title : '';
+};
+
+const RichLinkBody: React.FC<{ richLink: RichLink }> = ({ richLink }) => {
+    const linkText =
+        richLink.cardStyle === 'letters'
+            ? `${richLink.headline} | Letters `
+            : richLink.headline;
+
+    const imageCardStyles = ['news', 'letters', 'media', 'feature'];
+    const showImage =
+        richLink.thumbnailUrl &&
+        imageCardStyles.indexOf(richLink.cardStyle) >= 0;
+    const isPaidContent = richLink.tags
+        ? richLink.tags.filter(t => t.id === 'tone/advertisement-features')
+              .length > 0
+        : false;
+    const isOpinion = richLink.cardStyle === 'comment';
+    const mainContributor = getMainContributor(richLink.tags);
+    return (
+        <a className={cx(richLinkLink)} href={richLink.url}>
+            <div className={richLinkTopBorder(richLink.pillar)} />
+            {showImage && (
+                <div>
+                    <img
+                        src={richLink.thumbnailUrl}
+                        alt="link thumbnail image"
+                    />
+                </div>
+            )}
+            <div className={richLinkElements}>
+                <div className={richLinkTitle}>
+                    {isOpinion && (
+                        <div className={quote(richLink.pillar)}>
+                            <Quote />
+                        </div>
+                    )}
+                    {linkText}
+                </div>
+                {isOpinion && (
+                    <div className={cx(byline, textColour(richLink.pillar))}>
+                        {mainContributor}
+                    </div>
+                )}
+                {richLink.starRating && (
```

Comment:
I ask because react is counterintuitive here, it renders _some_ falsy values, such as `0`, so unless `richLink.starRating` is explicitly `false` or another clearly falsy value, then it will always render the component.

Reply:
👍